src/open_inwoner/openklant/services.py

```diff
@@ -1404,11 +1407,55 @@ def retrieve_question(
             q for q in all_questions if q.question_kcm_uuid == question_uuid
         )
 
-        # TODO
-        # should return (Question, zaak_with_api_group); the latter is left out until a
-        # standard for linking klantcontact + zaak is agreed upon
-        # https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/issues/808#issuecomment-2357637675
-        return self._build_question_dto(question_ok2=question, user=user), None
+        zaak_with_api_group = None
+
+        # fetch onderwerp_object linked to klantcontact
+        onderwerp_objecten = self.client.onderwerp_object.list()["results"]
+        onderwerp_object = next(
+            o
+            for o in onderwerp_objecten
+            if o["klantcontact"]["uuid"] == question.question_kcm_uuid
+        )
+
+        # fetch zaken for user
+        groups = list(
+            ZGWApiGroupConfig.objects.filter(
+                klant_backend=KlantenServiceType.OPENKLANT2.value
+            )
+        )
+        proxy = MultiZgwClientProxy([group.zaken_client for group in groups])
+        proxy_response = proxy.fetch_cases(user_bsn=user.bsn)
+        truthy_responses = proxy_response.truthy_responses
+
+        if (case_count := len(truthy_responses)) == 0:
+            logger.error(
+                "Unable to find matched contactmomenten zaak with OpenKlant2 backend"
+            )
+            return self._build_question_dto(question_ok2=question, user=user), None
+
+        # find zaak_with_api_group which is linked to the question
+        for response in truthy_responses:
+            zaken, client = response.result, response.client
+            for zaak in zaken:
+                if (
+                    not zaak.identificatie
+                    == onderwerp_object["onderwerpobjectidentificator"]["objectId"]
+                ):
```

Comment:
Taking an advance on #1614 :
```suggestion
                if (
                    zaak.identificatie
                    != onderwerp_object["onderwerpobjectidentificator"]["objectId"]
                ):
```